R/write.R

```diff
@@ -156,3 +147,55 @@ get_split_path <- function(path, ind) {
     tools::file_ext(path)
   )
 }
+
+#' Function to export data frame to xpt files ensuring each file does not exceed the maximum specified size in GB
+#'
+#' @param df A data frame to write.
+#' @param max_size_gb Maximum size in GB of the exported files.
+#' @param file_prefix Name of each exported file.
+#'
+#' @noRd
+
+export_to_xpt <- function(.df, path, max_size_gb, file_prefix) {
+  # Convert GB to bytes
+  max_size_bytes <- max_size_gb * 1000^3
+
+  total_rows <- nrow(.df)
+  chunk_counter <- 1
+  row_start <- 1
+
+  while (row_start <= total_rows) {
+    # Binary search to find the maximum number of rows that fit within the size limit
+    low <- row_start
+    high <- total_rows
+    best_fit <- row_start
+
+    while (low <= high) {
+      mid <- floor((low + high) / 2)
+      temp_file <- tempfile()
+      write_xpt(.df[row_start:mid, ], temp_file)
+      file_size <- file.info(temp_file)$size
+
+      if (file_size <= max_size_bytes) {
+        best_fit <- mid
+        low <- mid + 1
+      } else {
+        high <- mid - 1
+      }
+
+      unlink(temp_file) # Clean up the temporary file
+    }
+
+    # Write the best fitting chunk to the final file
+    file_name <- sprintf("%s%d.xpt", file_prefix, chunk_counter)
+    path <- sub("\\/[^/]*$", "/", path)
```

Comment:
Getting the directory path using string manipulation fails in Windows machines because in Windows the directory separators are different. I just pushed a commit to fix this which will eventually fix the CI checks.